src/rpc/methods/eth/types.rs

```diff
@@ -516,21 +516,41 @@ impl Default for TraceResult {
 
 #[derive(PartialEq, Default, Serialize, Deserialize, Debug, Clone, JsonSchema)]
 #[serde(rename_all = "camelCase")]
-pub struct EthBlockTrace {
+pub struct EthTrace {
     pub r#type: String,
     pub subtraces: i64,
     pub trace_address: Vec<i64>,
     pub action: TraceAction,
     pub result: TraceResult,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub error: Option<String>,
+}
+
+#[derive(PartialEq, Default, Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct EthBlockTrace {
+    #[serde(flatten)]
+    pub trace: EthTrace,
```

Comment:
Thanks for refactoring this! I hope this `flatten` attribute won't cause too much trouble when exporting our JSON OpenRPC API.

Reply:
When you say `JSON OpenRPC API` does it mean api result in compare to lotus API?.

Reply:
We can export an `OpenRPC` JSON document using `forest-tool shed openrpc --path v1`. For the sake of convergence, we aim to stay close to what Lotus generates.
One of our upcoming goal is to establish a common node API based on a shared specification.